integration/control_connection_test.rb

```diff
@@ -0,0 +1,71 @@
+# encoding: utf-8
+
+#--
+# Copyright 2013-2016 DataStax, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#++
+
+require File.dirname(__FILE__) + '/integration_test_case.rb'
+
+class ControlConnectionTest < IntegrationTestCase
+
+  def self.before_suite
+    @@ccm_cluster = CCM.setup_cluster(1, 2)
+  end
+
+  def self.after_suite
+    CCM.remove_cluster(@@ccm_cluster.name)
+  end
+
+  def remove_peer_info(info)
+    # Make sure to only connect to node1, as each node has its own peers table
+    allowed_ips = ['127.0.0.1']
+    round_robin = Cassandra::LoadBalancing::Policies::RoundRobin.new
+    whitelist = Cassandra::LoadBalancing::Policies::WhiteList.new(allowed_ips, round_robin)
+    cluster = Cassandra.cluster(load_balancing_policy: whitelist)
+    session = cluster.connect
+
+    session.execute("DELETE #{info} FROM system.peers WHERE peer = '127.0.0.2'")
+    result = session.execute("SELECT #{info} FROM system.peers WHERE peer = '127.0.0.2'").first
+    assert_nil result["#{info}"]
+
+    cluster.close
+  end
+
+  # Test for null columns in peer
+  #
+  # test_missing_peer_columns tests that the control connection ignores any peers which have missing peer columns.
+  # Using a simple 2-node cluster, it first removes one of the peer columns of node2 from node1's system.peers
+  # table. It then uses node1 explicitly as the control connection and verifies that node2 has not been used as a host.
+  #
+  # @since 2.1.7
+  # @jira_ticket RUBY-255
+  # @expected_result Node2 should not be used as a host
+  #
+  # @test_assumptions A 2-node Cassandra cluster.
+  # @test_category control_connection
+  #
+  def test_missing_peer_columns
+    peer_info = ['host_id', 'data_center', 'rack', 'rpc_address', 'tokens']
```

Comment:
Since you are progressively clearing attributes but never restoring them, the test isn't accurate. For example, let's say the driver filtered out peers based on nil host_id but not the other attributes. The test would erroneously pass because host_id is cleared first, making the peer drop out for the rest of the iterations.